decentraland/helpers/storage.js

```diff
@@ -7,47 +8,73 @@ const MY_ASSETS_LIST = "MY_ASSETS_LIST";
 
 export const storeEphemeralAccount = async account => {
   const { privateKey } = account;
-  await _storeData(EPHEMERAL_ACCOUNT_PRIV_KEY, privateKey);
+  await _storeData(EPHEMERAL_ACCOUNT_PRIV_KEY, privateKey, true);
 };
 
 export const retrieveEphemeralAccount = async () => {
   let account = null;
 
-  const privateKey = await _retrieveData(EPHEMERAL_ACCOUNT_PRIV_KEY);
+  const privateKey = await _retrieveData(EPHEMERAL_ACCOUNT_PRIV_KEY, true);
   if (privateKey != null) account = createFromPrivateKey(privateKey);
 
   return account;
 };
 
 export const storeMyAssets = async myAssets => {
-  const strMyAssets = JSON.stringify(myAssets);
-  await _storeData(MY_ASSETS_LIST, strMyAssets);
+  const strMyAssets = _fromJson(myAssets);
+  await _storeData(MY_ASSETS_LIST, strMyAssets, false);
 };
 
 export const retrieveMyAssets = async () => {
-  const strMyAssets = await _retrieveData(MY_ASSETS_LIST);
-  const myAssets = JSON.parse(strMyAssets);
+  const strMyAssets = await _retrieveData(MY_ASSETS_LIST, false);
+  const myAssets = _toJson(strMyAssets);
   return myAssets;
 };
 
+const _fromJson = json => {
+  try {
+    return JSON.stringify(json);
+  } catch {
+    throw Error(`Unable to parse JSON object to a String.`);
```

Comment:
This would be converting an object to a JSON string

Reply:
Done.

Reply:
Thanks! Resolving